src/main/java/com/microsoft/sqlserver/jdbc/AuthenticationJNI.java

```diff
@@ -157,9 +157,9 @@ private static native FedAuthDllInfo ADALGetAccessTokenForWindowsIntegrated(Stri
             String servicePrincipalName, String clientConnectionId, String clientId, long expirationFileTime,
             java.util.logging.Logger log);
 
-    static native byte[] DecryptColumnEncryptionKey(String masterKeyPath, String encryptionAlgorithm,
+    static synchronized native byte[] DecryptColumnEncryptionKey(String masterKeyPath, String encryptionAlgorithm,
             byte[] encryptedColumnEncryptionKey) throws DLLException;
 
-    static native boolean VerifyColumnMasterKeyMetadata(String keyPath, boolean allowEnclaveComputations,
+    static synchronized native boolean VerifyColumnMasterKeyMetadata(String keyPath, boolean allowEnclaveComputations,

```

Comment:
Do both these functions call the same problematic handle getter? If so, don't we need to prevent both of them from calling the dll function at the same time? Might need an Object lock here instead of synchronized functions.